From 94b91f723a23d06cc62800b264d021e81cb16432 Mon Sep 17 00:00:00 2001
From: Kai Uwe Broulik <kde@privat.broulik.de>
Date: Mon, 15 Jan 2018 10:32:40 +0100
Subject: [ConfigModel] Set QML context on ConfigModule

This will have the ConfigModule use the same engine as we do, ensuring we use the correct QtQuick Controls style.

BUG: 388766

Differential Revision: https://phabricator.kde.org/D9785
---
 src/plasmaquick/configmodel.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/plasmaquick/configmodel.cpp b/src/plasmaquick/configmodel.cpp
index 502edfc..5f6f47a 100644
--- a/src/plasmaquick/configmodel.cpp
+++ b/src/plasmaquick/configmodel.cpp
@@ -269,6 +269,12 @@ QVariant ConfigModel::data(const QModelIndex &index, int role) const
             if (!cm) {
                 qWarning() << "Error creating KCM object from plugin" << loader.fileName();
             }
+
+            if (QQmlContext *ctx = QQmlEngine::contextForObject(this)) {
+                // assign the ConfigModule the same QML context as we have so it can use the same QML engine as we do
+                QQmlEngine::setContextForObject(cm, ctx);
+            }
+
             d->kcms[pluginName] = cm;
             return QVariant::fromValue(cm);
         }
-- 
cgit v0.11.2

